Sources/VitaminUIKit/Components/Chip/README.md

```diff
@@ -0,0 +1,222 @@
+- Input
+- Single Choice
+- Filter
+
+These different variants are represented by the `VitaminChipVariant` enum, wifth 4 cases:
+- `.action`
+- `.input`
+- `.singleChoice`
+- `.filter`
+
+Each of these cases have associated values to specify additional properties only relevant for each variant.
+
+Default value is `.filter(state: .unselected, badge: nil)` (see below).
+
+
+#### Action Variant
+The action chip has a mandatory icon on the left hand side of the `VitaminChip`'s text.
+This icon will be colored and resized to 20px x 20px.
+You can specify it via the icon associated value of the variant
+
+```swift
+import Vitamin
+
+// This chip will have a settings icon
+let chip = VitaminChip(text: "test", variant: .action(icon: Vitamix.Line.System.settings.image))
+
+// You can change the icon
+chip.variant = .action(icon: Vitamix.Line.Map.mapPin.image)
+```
+
+#### Input Variant
+The input variant is a chip that you can dynamically add after the user has typed and validated an input.
+It always has a close icon on the right hand side of the text, to indicate it can be dismissed.
+It can have either an icon or an image on the left hand side of the text.
+
+If you provide an icon, it wil be colored and resized to 20px x 20px.
+If you provide an image, it will be ronded and resized to 20px x 20px.
+If you provide both, only icon will be taken in account.
+
+You can specify icon or image via the `icon` or `image` associated values ofthe variant.
+
+⚠️ The dismissal of the chip is not iplemented in this library, it is up to you to dismiss the chip when clicked (see #action below)
+
+```swift
+import Vitamin
+
+let chip = VitaminChip(text: "test", variant: .input())
+chip.variant = .input(icon: Vitamix.Line.System.settings.image)
+
+let anotherChip = VitaminChip(text: "test", variant: .input())
+chip.variant = .input(image: VitaminAssets.Placeholder.landscape.image)`
+
+let chipWithImage = VitaminChip(text: "test", variant: .input(image: VitaminAssets.Placeholder.landscape.image))
+
+let chipWithIcon = VitaminChip(text: "test", variant: .input(icon: Vitamix.Line.System.settings.image))
+```
+
+
+
+#### Single choice variant
+The single choice variant is meant to be used in a group of `VitaminChip`, in which only one chip could be selected at a time.
+
+You can provide a `VitaminChipVariantState` in the state associated value of the variant.
+It is an enum with two cases:
+- `.selected`
+- `.unselected`
+
+To change the state, you have to change the whole variant property with a property new `VitaminChipVariant` with the opossite state.
+To help you in that, the `VitaminChipVariantState` provides a `toggle()` method that will return the opposite state.
+
+The change of state on click is handled by the component : each time you touch the component, it will toggle to opposite state.
+
+⚠️ The change of state of other single choice `VitaminChip`in the same group is not handled by this library, it is up to you to ahndle that when clicked (see #action).
```

Comment:
In the documentation, it is indicated that when you use SingleChoice, only one of them can be selected at a time.
But I thought it was too complex to implement it in the component, because that would mean that the chip has a reference on others chips.
I was just trying to explain that it's app responsability to ensure that, and that the change of state of others chips (from .selected to .unselected) could be triggered in the action callback if needed